common/tx-batch-module/src/lib.rs

```diff
@@ -183,7 +183,7 @@ pub trait TxBatchModule {
         }
 
         let max_batch_size = self.max_tx_batch_size().get();
-        if tx_batch.len() == max_batch_size {
+        if tx_batch.len() >= max_batch_size {
```

Comment:
why?

Reply:
If max_tx_batch_size is set to a smaller value than the current batch size, is_batch_full would never return true for this batch as it performs an equality check.